### PR TITLE
Replace ::String with ::AbstractString

### PR DESCRIPTION
**Pull request by nkottary**

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ stored in a subdirectory corresponding to the registry's UUID.
 Maintains a dictionary `registries` which maps registry repository URLs to UUIDs.
 """
 struct RegistryCache
-    path::String
+    path::AbstractString
```
[the remark this answers is not in the document]
Comment (nkottary) [reply src/types.jl:10]: `SubString{String}` might get passed as path to the constructor.

[the remark this answers is not in the document]
Comment (GunnarFarneback) [reply src/types.jl:10]: It just gets converted to a `String` in that case.
```
julia> struct Foo
           s::String
       end

julia> foo = Foo(SubString{String}("a"))
Foo("a")
```

[the remark this answers is not in the document]
Comment (ericphanson) [reply src/types.jl:10]: Yeah, I think we should just change in the functions and not the structs

[the remark this answers is not in the document]
Comment (nkottary) [reply src/types.jl]: Fixed in https://github.com/JuliaRegistries/RegistryTools.jl/pull/72/commits/356bf631b1845c388ffebc1efac4bdb07bc4ae31

**Pull request merged**